### DISCUSSION TITLE
Add simple channel routing

### DISCUSSION THREAD

**Discussion by jeffmikels** [ExistentialAudio/BlackHole#66] [Feature Requests]
As an expansion on #13, I'd like to have very basic routing from input channels to output channels. For example, I'd like to send audio to channels 1 and 2, a click track to channel 3, and LTC timecode to channel 4. This would be killer, and I'm willing to help. Can you point me to the right starting point in your code for doing something like this?

**Comment by devinroth** [comment]
Hey Jeff. I'm not exactly sure what you are asking. Could you elaborate please?

**Comment by jeffgca** [comment]
This issue seems to be what I want to request as well, so I'll try to elaborate but @jeffmikels if this isn't what you mean please let me know and I'll file another issue.

I own a DJ mixer that has a built in sound card with 12 inputs, and the summed master output of the mixer is routed to channels 11 & 12. I ( along with every other DJ on the planet right now ) want to use OBS to stream sets and OBS currently only supports channels 1 & 2 from an external sound card.

[added] OBS has so far declined to add routing on macOS but there is an ['idea'](https://ideas.obsproject.com/posts/199/multichannel-audio-interface-support-for-mac) logged. OBS support this via an ASIO component on Windows.

The most mentioned workaround for this issue is to spend $100 on [loopback](https://rogueamoeba.com/loopback/). I already own [Audio Hijack](https://rogueamoeba.com/audiohijack/) so my workaround is to use AH to route the mixers 11&12 input channels to Blackhole's virtual 1&2 channels. 

It would be great to be able to route audio from my mixer's hardware inputs on channels 11 & 12 to BH's 1&2 outputs in a way that doesn't involve additional software. loopback and Jack have UI for this sort of thing, perhaps @jeffmikels has experience building UI on macOS adn can contribute this?

**Comment by devinroth** [comment]
Just to make sure I understand.

You want to output to channel 11-12 of BlackHole so that those channels come out of BlackHole's input channels 1-2. Right?

**Comment by jeffgca** [comment]
Nope, and sorry for any confusion. I want something that is probably out of scope for blackhole unless you add a UI panel - I want to route audio coming from an external audio in on channels 11&12 to blackhole channels 1&2.

**Comment by devinroth** [comment]
Makes sense now. 

I’m unsure if I want to roll that into BlackHole but instead make a separate utility for that. I like to keep stuff modular.

**Comment by jeffgca** [comment]
@devinroth I think I agree, Blackhole currently has a very discrete feature set. If you're interested in creating some sort of complementary routing utility I would be interested in contributing.

**Comment by devinroth** [comment]
Thanks for you willingness to help @canuckistani. I will definitely keep it in mind.

**Comment by krono-i2** [comment]
I’m also interested in this routing functionality.
Let me know if you need help anyway. Code and test.

Ivan
